src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -128,33 +133,19 @@ public Result getFirstResult() {
         return null;
     }
 
-    // TODO NR, SE: remove redundant setter after relationship change on client. Currently we need two deserializing setters for "result" (client) and "results" (server)
-    @JsonProperty(value = "result", access = JsonProperty.Access.WRITE_ONLY)
-    public void setResult(Result result) {
-        this.setResults(result);
-    }
-
-    // TODO Ruscher, Entholzer: refactor to addResult
     /**
-     * custom setter that supports the migration from 1...1 to 1...* in the submission->result(s) relationship
-     * Will be refactore in the future
-     * @param result the result that should be added, in case this is null, an empty list will be used instead
+     * Used as a setResult method, as typically the latest result is used
+     *
+     * @param result
      */
-    @JsonProperty(value = "results", access = JsonProperty.Access.WRITE_ONLY)
-    public void setResults(Result result) {
-        if (result == null) {
-            // clear the list of results
-            this.results = new ArrayList<>();
-        }
-        else {
-            // addResult
-            this.results.add(result);
-        }
+    public void addResult(Result result) {
+        this.results.add(result);
+        // At the moment only one result in results is allowed
+        assert results.size() < 2;
     }
 
-    // TODO Ruscher, Entholzer: refactor to setResults
-    @JsonIgnore()
-    public void setResultsList(List<Result> results) {
+    @JsonProperty(value = "results", access = JsonProperty.Access.WRITE_ONLY)
+    public void setResults(List<Result> results) {
```

Comment:
It looks like this is mostly/only used for clearing `results`. Wouldn't it make more sense to add a dedicated clear method? (This way we also don't allocate a new ArrayList each time we just want to throw the old results away.)

Reply:
Yes for now its only used for that, but the method will be used for the upcoming multi correction feature, therefore we should keep the name for now.

Reply:
I was more trying to suggest creating a new method which is just used for clearing. Or will clearing get obsolete once we have multi correction?